Gem/Code/Source/Components/UI/HUDComponent.cpp

```diff
@@ -12,13 +12,16 @@
 #include <AzCore/Serialization/SerializeContext.h>
 #include <AzCore/Serialization/EditContext.h>
 
+#if AZ_TRAIT_CLIENT
 #include <LyShine/Bus/UiCanvasBus.h>
 #include <LyShine/Bus/UiTextBus.h>
+#endif
 
 namespace MultiplayerSample
 {
     void HUDComponent::Activate()
     {
+#if AZ_TRAIT_CLIENT
```

Comment:
this entire class seems like something you'd only want on the client - is there a way to do that?

Reply:
or do you need it on the server so that it can tell the client what components to spawn on the player?

Reply:
Correct. Unfortunately there's no concept of split assets so we need to maintain the component and its serialization elements for the time being. Long term, it would be great if we had no macro usage in this files and excluded this via cmake file list.